pandas/tests/extension/integer/test_integer.py

```diff
@@ -143,283 +129,80 @@ def test_array_type_with_arg(self, data, dtype):
-        # ndarray & other series
-        op = all_arithmetic_operators
-
-        s = pd.Series(data)
-        other = np.ones(len(s), dtype=s.dtype.type)
-        self._check_op(s, op, other, exc=TypeError)
-
-    def test_arith_coerce_scalar(self, data, all_arithmetic_operators):
-
-        op = all_arithmetic_operators
-        s = pd.Series(data)
-
-        other = 0.01
-        self._check_op(s, op, other)
-
-    @pytest.mark.parametrize("other", [1., 1.0, np.array(1.), np.array([1.])])
-    def test_arithmetic_conversion(self, all_arithmetic_operators, other):
-        # if we have a float operand we should have a float result
-        # if if that is equal to an integer
-        op = self.get_op_from_name(all_arithmetic_operators)
+                # combine method result in 'biggest' (int64) dtype
+                expected = expected.astype(s.dtype)
+                pass
+            if (op_name == '__rpow__') and isinstance(other, pd.Series):
+                # TODO pow on Int arrays gives different result with NA
+                # see https://github.com/pandas-dev/pandas/issues/22022
+                result = result.fillna(1)
+
+            self.assert_series_equal(result, expected)
+        else:
+            with pytest.raises(exc):
+                op(s, other)
 
-        s = pd.Series([1, 2, 3], dtype='Int64')
-        result = op(s, other)
-        assert result.dtype is np.dtype('float')
+    def _check_divmod_op(self, s, op, other, exc=None):
+        super(TestArithmeticOps, self)._check_divmod_op(s, op, other, None)
 
+    @pytest.mark.skip(reason="intNA does not error on ops")
     def test_error(self, data, all_arithmetic_operators):
-        # invalid ops
-
-        op = all_arithmetic_operators
-        s = pd.Series(data)
-        ops = getattr(s, op)
-        opa = getattr(data, op)
-
-        # invalid scalars
-        with pytest.raises(TypeError):
-            ops('foo')
-        with pytest.raises(TypeError):
-            ops(pd.Timestamp('20180101'))
-
-        # invalid array-likes
-        with pytest.raises(TypeError):
-            ops(pd.Series('foo', index=s.index))
-
-        if op != '__rpow__':
-            # TODO(extension)
-            # rpow with a datetimelike coerces the integer array incorrectly
-            with pytest.raises(TypeError):
-                ops(pd.Series(pd.date_range('20180101', periods=len(s))))
-
-        # 2d
-        with pytest.raises(NotImplementedError):
-            opa(pd.DataFrame({'A': s}))
-        with pytest.raises(NotImplementedError):
-            opa(np.arange(len(s)).reshape(-1, len(s)))
+        # other specific errors tested in the integer array specific tests
+        pass
 
 
 class TestComparisonOps(BaseInteger, base.BaseComparisonOpsTests):
 
-    def _compare_other(self, s, data, op_name, other):
-        op = self.get_op_from_name(op_name)
+    def check_opname(self, s, op_name, other, exc=None):
```

Comment:
things like this are code smell. you have not idea what is going on.

Reply:
I can add a comment, but this is to override the default of `exc` in the parent class (which is an identical pattern as in the decimal tests ànd as in the original integer tests that you added, see eg `_check_divmod_op` in `TestArithmeticOps` a bit above (in the deleted lines in this diff))